docs/data/pages.ts

```diff
@@ -105,6 +105,39 @@ const pages: MuiPage[] = [
           { pathname: '/x/react-data-grid/pivoting', title: 'Pivoting 🚧', plan: 'premium' },
         ],
       },
+      {
+        pathname: '/x/react-data-grid/server-side-data-group',
+        title: 'Server-side data 🚧',
+        plan: 'pro',
```

Comment:
Are we sure about this one?

```suggestion
```

I can't find the Notion discussion back about it.

Reply:
This is an interesting discussion. 👍 

The decision was to not support the MIT features (i.e filtering, sorting, and pagination) with the data source (see existing server-side features section [here](https://mui.com/x/react-data-grid/server-side-data/#usage)), and I tried to cover under this section only those sub-sections that correspond to the server-side features bound to the data source. You are right, it might be confusing though as most users would think all the server-side features (including MIT) might be covered in the server-side data section.

Two possibilities we could go with:
1. Go with the current assumption and alias the MIT corresponding features to the original sections and vice versa like this for example:
   - **Server-side filtering (MIT)** linked to https://mui.com/x/react-data-grid/filtering/server-side/
   - **Server-side sorting (MIT)** linked to https://mui.com/x/react-data-grid/sorting/#server-side-sorting
   - **Server-side pagination (MIT)** linked to https://mui.com/x/react-data-grid/pagination/#server-side-pagination
2. Or we could move all the server-side features (MIT and Pro) under the **server-side data** root section, create new sub-sections for filtering, sorting, and pagination, and have them cover both MIT (without data source) and Pro (with data source) sub-sections. The original sections for filtering, sorting, and pagination (linked in point no. 1) will then either have to be removed or aliased to the new sections.

What do you think?

@mui/xgrid Feel free to hop in